File: 2024/august/java/06.java
class Solution {
    private int [] f;
    public int minimumPushes(String word) {
        f = new int[26];
        for (var ch : word.toCharArray()){
            f[ch-'a']++;
        }

        Integer [] id = new Integer [26];
        for (int i = 0; i < 26; ++i) id[i] = i;


      Arrays.sort(id, (a, b) -> Integer.compare(f[b], f[a]));
      int k = 0;
      int current = 1;
      int ans = 0;
      for (var cur : id){
        ++k;
        if (k == 9){
            ++current;
            k = 1;
        }
        ans+= f[cur] * current;
       // System.out.println((char)(cur+'a') + "  "+ f[cur] + "  "+current);
      }
      return ans;
        
    }
}

File: 2024/august/javaScript/06.js
/**
 * @param {string} word
 * @return {number}
 */
var minimumPushes = function(word) {
      let f = Array.from({length: 26} , ()=>0);

        const value =function (a){
          return a.charCodeAt(0)-97;
        };
        for (const ch of word){
          f[value(ch)]++;
        }

        const id = Array.from({length : 26} , (_ , i)=> i);
      id.sort((a, b) => f[b]-f[a]);
      let k = 0;
      let current = 1;
      let ans = 0;
      for (const cur of id){
        ++k;
        if (k == 9){
            ++current;
            k = 1;
        }
        ans+= f[cur] * current;
      }
      return ans;
};
